src/lib/gameService.ts

```diff
@@ -12,41 +13,172 @@ export class GameService {
+		// its "rollDice" function because you would always roll all of the dice.
+		// Would need a Game to be passed to it though: "diceRoller.rollAllDice(game.initialDiceCount, game)"";
+		this.chooseStartingPlayer(numPlayers, game);
+		this.updateGameState(GameState.InProgress, game);
+
+		this.repository.saveGame(game);
+
+		// throw new Error('not implemented');
 	}
 
-	public getGame(playerToken: string): Promise<GameDto> {
+	public async getGame(playerToken: string): Promise<GameDto> {
+		// // Currently not sure where to convert to OpponentDtos for left and right players.
+		// const gameCode = this.getGameCode(playerToken);
+		// const game: Game | undefined = await gameRepository.getGame(gameCode);
+		// if (!game) {
+		// 	throw new Error('Game is undefined');
+		// }
+		// const players: PlayerDto[] = await this.getPlayers(gameCode);
+		// const numPlayers = players.length;
+		// const playerIndex: number = game.players.findIndex((player) =>
+		// 	player.name.includes(this.getPlayerName(playerToken))
+		// );
+		// const leftPlayerIndex: number = (playerIndex - 1 + numPlayers) % numPlayers;
+		// const rightPlayerIndex: number = (playerIndex + 1) % numPlayers;
+
+		// const playerToLeft: PlayerDto = players[leftPlayerIndex];
+		// const playerToRight: PlayerDto = players[rightPlayerIndex];
+		// const playerDice: number[] = game.players[playerIndex].dice;
+		// let playerTurn: 'player' | 'left' | 'right';
+		// if (game.currentPlayer == playerIndex) {
+		// 	playerTurn = 'player';
+		// } else if (playerIndex >= leftPlayerIndex) {
+		// 	playerTurn = 'left';
+		// } else {
+		// 	playerTurn = 'right';
+		// }
+
+		// const gameDetails: GameDto = {
+		// 	left: playerToLeft,
+		// 	right: playerToRight,
+		// 	dice: playerDice,
+		// 	turn: playerTurn
+		// };
+
+		// return gameDetails;
+
 		throw new Error('not implemented');
 	}
 
-	public async placeBid(bid: Bid, playerToken: string): Promise<void> {}
+	public async placeBid(bid: Bid, playerToken: string): Promise<void> {
+		const gameCode = this.getGameCode(playerToken);
+		const game: Game | undefined = await gameRepository.getGame(gameCode);
+		if (!game) {
+			throw new Error('Game is undefined');
+		}
 
-	public async challengeBid(bid: Bid, playerToken: string): Promise<void> {}
+		throw new Error('not implemented');
+	}
+
+	public async challengeBid(bid: Bid, playerToken: string): Promise<void> {
+		throw new Error('not implemented');
+	}
 
-	public async peekDice(playerToken: string): Promise<void> {}
+	public async peekDice(playerToken: string): Promise<void> {
+		throw new Error('not implemented');
+	}
 
-	private generateCode() {
+	public generateCode() {
 		const firstPart = (Math.random() * 46656) | 0;
 		const secondPart = (Math.random() * 46656) | 0;
 		return ('000' + firstPart.toString(36)).slice(-3) + ('000' + secondPart.toString(36)).slice(-3);
 	}
+
+	private getGameCode(playerToken: string): string {
```

Comment:
This will probably always be called alongside getPlayerName(), they could be combined to return both at once:
```
public splitPlayerToken(playerToken: string): {gameCode: string, playerCode: boolean} {
...
}
```

The consumer would call it like so:
```
let {gameCode, playerCode} = splitPlayerToken(playerToken);
```